src/validationinterface.h

```diff
@@ -34,6 +35,8 @@ class CValidationInterface {
     virtual void Inventory(const uint256 &hash) {}
     virtual void ResendWalletTransactions(int64_t nBestBlockTime) {}
     virtual void BlockChecked(const CBlock&, const CValidationState&) {}
+    virtual void GetScriptForMining(CScript &script) {};
```

Comment:
Why not:

    CScript GetScriptForMining();

Yes - you'll need to specify a combiner for the return argument, but right now this issue ("what if there are multiple subscribers that return a value") is hidden.